### PR TITLE
[skip ci] Update minimum supported compiler levels and fix documentation links

### PR DESCRIPTION
**Pull request by 0xdaryl**
Also add missing platforms (Arm Linux and macOS).

Removed XLC for Linux as Eclipse OMR does not have a means of evaluating
the latest compiler versions as part of its CI builds.

Issue #7359

### CODE REVIEW

**Comment by 0xdaryl** [comment]
@mstoodle please review.

@ThanHenderson FYI

Note that this PR updates the minimum supported compiler section only.  More work is required to clean up the rest of the document on what features are actually supported (or not).

**Comment by ThanHenderson** [comment]
Great. Thanks.

> More work is required to clean up the rest of the document on what features are actually supported (or not)

Now that the versions are documented, I can run through that list and shuffle around the features that are now supported. iirc from my initial investigation, all the compilers save for xlC are C++11 feature complete.

File: doc/SupportedC++Features.md
Comment (ThanHenderson) [inline doc/SupportedC++Features.md:35]: Are these versions for Linux and AIX XLC compilers correct? From the [conversation](https://github.com/eclipse/omr/issues/7359#issuecomment-2161689154) in the issue, I think 13.1.3 is the minimum version.

**Comment by 0xdaryl** [comment]
> Now that the versions are documented, I can run through that list and shuffle around the features that are now supported. iirc from my initial investigation, all the compilers save for xlC are C++11 feature complete.

While it is a nice-to-have for the 0.3.0 release, I don't think that it is as "urgent" as updating and documenting the minimum compiler levels supported.  If that is something you'd be able to do by the end of the week then we can wait (let me know).  Otherwise, once this PR is merged I will publish the 0.3.0 release.

File: doc/SupportedC++Features.md
Comment (0xdaryl) [reply doc/SupportedC++Features.md:35]: Yes, you're right.  Thanks for spotting that.

**Comment by ThanHenderson** [comment]
Given https://github.com/eclipse-openj9/openj9/pull/20304. The features moved from the supported features list in PR https://github.com/eclipse/omr/pull/7369 should be moved back to the unsupported features list. Could you add that here? If not, I'll open another PR.

That PR moved the features that were labelled as blocked by only `gcc 4.4.7` to the supported features list; it is clear now though that it is XLC on z/OS that is the blocker.

**Comment by 0xdaryl** [comment]
I want to keep this PR just for the compiler levels.  If there are changes to the supported features please contribute in a separate PR.

**Comment by 0xdaryl** [comment]
@kevindean12 : please confirm you have no issues with this proposal by approving

**Comment by mstoodle** [comment]
@dibyendumajumdar I would guess that you're using at least these levels, but since we're moving things forward, I just thought I'd check in with you for awareness at the very least...

**Review by kevindean12 — APPROVED** [review]

**Review by mstoodle — APPROVED** [review]
lgtm, thanks @0xdaryl

**Comment by mstoodle** [comment]
not sure if this will work, but worth a try

[skip ci]

**Pull request merged**
